lib/collection/src/collection/mod.rs

```diff
@@ -112,6 +115,7 @@ impl Collection {
                 channel_service.clone(),
                 update_runtime.clone().unwrap_or_else(Handle::current),
                 search_runtime.clone().unwrap_or_else(Handle::current),
+                optimizer_cpu_budget.clone(),
```

Comment:
Does this mean that all nodes will have same cpu budget for optimizations? I am thinking maybe we should let each node have its own config

Reply:
I think you meant 'that all collections will have the same CPU budget for optimizations'. 

CPU optimizations are on node/instance level. So a node has a global CPU budget internally, which is shared across all collections.

We must share this budget this to prevent two collections consuming more CPU together than is available on the host.

_Once the public API and configuration changes are exposed as well:_ each node has the global budget configured in their own `config.yaml`. Concurrency across collections _can_ be controlled with the optimizer parameters at collection creation, such as `optimizers_config.max_optimization_threads` ([API](https://qdrant.github.io/qdrant/redoc/index.html#tag/collections/operation/get_collection)).

Do you think this is sufficient?

Reply:
Ah, right. I got it now. It's same budget but per instance, not per collection across instances. It makes much more sense this way

Reply:
Yes, that's another good way of saying it! :smile: